geniusyield-server-lib/src/GeniusYield/Server/Run.hs

```diff
@@ -0,0 +1,96 @@
+module GeniusYield.Server.Run (
+  runServer,
+) where
+
+import Control.Monad.Trans.Except (ExceptT (ExceptT))
+import Data.Aeson.Encode.Pretty (encodePretty)
+import Data.Version (showVersion)
+import Data.Yaml.Pretty qualified as Yaml
+import Fmt
+import GeniusYield.GYConfig
+import GeniusYield.HTTP.Errors
+import GeniusYield.Imports
+import GeniusYield.OrderBot.Adapter.Maestro (MaestroProvider (MaestroProvider))
+import GeniusYield.Providers (networkIdToMaestroEnv)
+import GeniusYield.Server.Api
+import GeniusYield.Server.Auth
+import GeniusYield.Server.Config (ServerConfig (..), coreConfigFromServerConfig, optionalSigningKeyFromServerConfig, serverConfigOptionalFPIO)
+import GeniusYield.Server.Constants (gitHash)
+import GeniusYield.Server.Ctx
+import GeniusYield.Server.ErrorMiddleware
+import GeniusYield.Server.RequestLoggerMiddleware (gcpReqLogger)
+import GeniusYield.Server.Utils
+import GeniusYield.Types
+import Network.Wai qualified as Wai
+import Network.Wai.Handler.Warp qualified as Warp
+import PackageInfo_geniusyield_server_lib qualified as PackageInfo
+import RIO hiding (Handler, logDebug, logErrorS, logInfo, logInfoS, onException)
+import RIO.ByteString qualified as B
+import RIO.ByteString.Lazy qualified as BL
+import RIO.Text.Lazy qualified as LT
+import Servant
+-- import Servant.PY (requests, writePythonForAPI)
+import Servant.Server.Experimental.Auth (AuthHandler)
+import Servant.Server.Internal.ServerError (responseServerError)
+import System.TimeManager (TimeoutThread (..))
+
+runServer ∷ Maybe FilePath → IO ()
+runServer mfp = do
+  serverConfig ← serverConfigOptionalFPIO mfp
+  menv ← networkIdToMaestroEnv (case scMaestroToken serverConfig of Confidential t → t) (scNetworkId serverConfig)
+  let optionalSigningKey = optionalSigningKeyFromServerConfig serverConfig
+      nid = scNetworkId serverConfig
+      coreCfg = coreConfigFromServerConfig serverConfig
+  -- writePythonForAPI (Proxy @MainAPI) requests "web/swagger/api.py"
+  withCfgProviders coreCfg "server" $ \providers → do
+    let logInfoS = gyLogInfo providers mempty
+        logErrorS = gyLogError providers mempty
+    logInfoS $ "GeniusYield server version: " +| showVersion PackageInfo.version |+ "\nCommit used: " +| gitHash |+ ""
+    BL.writeFile "web/swagger/api.json" (encodePretty geniusYieldAPISwagger)
```

Comment:
Is there a reason to export the JSON as well? I don't really see why we should have both JSON and YAML.

Reply:
If there is no need for JSON representation, we can get rid of it. I have done so in latest state of PR.